internal/service/kubernetes/node_group.go

```diff
@@ -2,470 +2,674 @@ package kubernetes
+				},
 			},
-			"name": {
-				Description:      "The name of the node group. Needs to be unique within a cluster.",
-				Type:             schema.TypeString,
-				Required:         true,
-				ValidateDiagFunc: validateResourceName,
-				ForceNew:         true,
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Computed ID of the node group. This is a combination of the cluster UUID and the node group name, separated with a `/`.",
+				Computed:            true,
 			},
-			"plan": {
-				Description: "The server plan used for the node group. You can list available plans with `upctl server plans`",
-				Type:        schema.TypeString,
-				Required:    true,
-				ForceNew:    true,
+			"labels": utils.LabelsAttribute("node_group", mapplanmodifier.RequiresReplace()),
+			"name": schema.StringAttribute{
+				MarkdownDescription: "The name of the node group. Needs to be unique within a cluster.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.LengthAtMost(resourceNameMaxLength),
+					stringvalidator.RegexMatches(resourceNameRegexp, fmt.Sprintf("name should only contain lowercase alphanumeric characters and dashes (a-z, 0-9, -). Name should not start or end with a dash. Regular expresion used to check validation: %s", resourceNameRegexp)),
+				},
 			},
-			"anti_affinity": {
-				Description: `If set to true, nodes in this group will be placed on separate compute hosts.
-				Please note that anti-affinity policy is considered "best effort" and enabling it does not fully guarantee that the nodes will end up on different hardware.`,
-				Type:     schema.TypeBool,
-				Optional: true,
-				Default:  false,
-				ForceNew: true,
+			"node_count": schema.Int64Attribute{
+				MarkdownDescription: "Amount of nodes to provision in the node group.",
+				Required:            true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(0),
+				},
 			},
-			"labels": {
-				Description: "Key-value pairs to classify the node group.",
-				Type:        schema.TypeMap,
-				Elem: &schema.Schema{
-					Type: schema.TypeString,
+			"plan": schema.StringAttribute{
+				MarkdownDescription: "The server plan used for the node group. You can list available plans with `upctl server plans`",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
 				},
+			},
+			"ssh_keys": schema.SetAttribute{
+				MarkdownDescription: "You can optionally select SSH keys to be added as authorized keys to the nodes in this node group. This allows you to connect to the nodes via SSH once they are running.",
+				ElementType:         types.StringType,
+				Optional:            true,
+				PlanModifiers: []planmodifier.Set{
+					setplanmodifier.RequiresReplace(),
+				},
+			},
+			"storage_encryption": schema.StringAttribute{
+				Computed: true,
```

Comment:
Description seems to be missing. Could also add note on how the default value is computed (inherited from cluster?)

Reply:
Added with the note